src/tests/ftest/util/server_utils.py

```diff
@@ -50,292 +47,352 @@ class ServerFailed(Exception):
+            "format": "SCM format required",
+            "normal": "DAOS I/O server.*started",
+        }
+        start_time = time.time()
+        start_msgs = 0
+        timed_out = False
+        while start_msgs != len(self._hosts) and not timed_out:
+            output = subprocess.get_stdout()
+            start_msgs = len(re.findall(patterns[self.mode], output))
+            timed_out = time.time() - start_time > self.timeout
+
+        if start_msgs != len(self._hosts):
+            err_msg = "{} detected. Only {}/{} messages received".format(
+                "Time out" if timed_out else "Error",
+                start_msgs, len(self._hosts))
+            print("{}:\n{}".format(err_msg, subprocess.get_stdout()))
+            raise CommandFailure(err_msg)
+
+        print("Started server in <{}> mode.".format(self.mode))
+        return True
+
+    def start(self, log_file=None):
+        """Start the server through the runner."""
+
+        self.set_server_yaml(log_file)
+
+        if self.runner.job.yaml.is_nvme():
+            self.mode = "format"
+            storage_prepare(self._hosts)
+            self.prep()
+
         try:
-            SESSIONS[setname].send_signal(signal.SIGINT)
-            time.sleep(5)
-            # get the stderr
-            error = SESSIONS[setname].stderr.read()
-            if SESSIONS[setname].poll() is None:
-                SESSIONS[setname].kill()
-            retcode = SESSIONS[setname].wait()
-            print(
-                "<SERVER> server start return code: {}\nstderr:\n{}".format(
-                    retcode, error))
-        except KeyError:
-            pass
-        raise ServerFailed("Server didn't start!")
-
-
-def stop_server(setname=None, hosts=None):
-    """Stop the daos servers.
-
-    Attempt to initiate an orderly shutdown of all orterun processes it has
-    spawned by sending a ctrl-c to the process matching the setname (or all
-    processes if no setname is provided).
-
-    If a list of hosts is provided, verify that all daos server processes are
-    dead.  Report an error if any processes are found and attempt to forcably
-    kill the processes.
+            self.runner.run()
+        except CommandFailure as details:
+            print("Failed to start server. ERROR: {}".format(details))
 
-    Args:
-        setname (str, optional): server group name used to match the session
-            used to start the server. Defaults to None.
-        hosts (list, optional): list of hosts running the server processes.
-            Defaults to None.
+        if self.runner.job.yaml.is_nvme():
+            storage_format(self._hosts)
 
-    Raises:
-        ServerFailed: if there was an error attempting to send a signal to stop
-            the processes running the servers or after sending the signal if
-            there are processes stiull running.
+        self.mode = "normal"
+        try:
+            self.check_subprocess_status(self.runner._process)
```

Comment:
(pylint-protected-access) Access to a protected member _process of a client class